### ISSUE TITLE
jquery.timeago library doesnt exists?

### ISSUE THREAD

**Issue by AlbertoP57** [YAFNET/YAFNET#536]
It seems that jquery.timeago.js is not used anymore, neither as package neither as .js file.

So, atm:
if (Host -> Host Settings -> Features -> Show relative time == true)
{
DisplayDateTime.cs returns time in GMT format in many places
}
else
{
DisplayDateTime.cs returns time in local format
}

Maybe are you planning to remove the:
Host -> Host Settings -> Features -> Show relative time
option? (my two cents: I agree)

**Comment by w8tcha** [comment]
Where does it shows no relative time? Can you describe where exactly?

**Comment by AlbertoP57** [comment]
Each:
<YAF:DisplayDateTime ...>

renders the class: DisplayDateTime,cs

or:

writer.Write(
ControlHtml,
this.Get<BoardSettings>().ShowRelativeTime
?
### this.AsDateTime.ToString("yyyy-MM-ddTHH:mm:ssZ", CultureInfo.InvariantCulture) ###
: ...

only when 
Host -> Host Settings -> Features -> Show relative time IS SET

if not needed anymore, remove che checkbox?
p.s.: TimeAgoForumModule.cs is unused?

**Comment by w8tcha** [comment]
Sorry i cant reproduce the issue the momentjs is still used in all the places. on which page is the relative time not working?

![fromt](https://user-images.githubusercontent.com/722575/75757387-8cc9ba00-5d32-11ea-8030-c306fcb6886f.png)

Why should the setting be removed it doesnt make any sense.

**Comment by AlbertoP57** [comment]
I repeat for the 3rd time:
Host -> Host Settings -> Features -> Show relative time 

![image](https://user-images.githubusercontent.com/18015672/75761491-75da9600-5d39-11ea-8f37-32971e9048d5.png)

then, wherever there is a reference to YAF:DisplayDateTime (for example in DisplayPost.ascx):
`
<YAF:DisplayDateTime id="DisplayDateTime" runat="server" 
DateTime='<%# this.DataRow["Posted"] %>'>
</YAF:DisplayDateTime>
`
you will get the GMT date:

![image](https://user-images.githubusercontent.com/18015672/75761804-fa2d1900-5d39-11ea-8f4a-72cd40042525.png)

I have wasted at least 2 hours of time. I know (almost) everything about this :)

**Comment by w8tcha** [comment]
Check the browser console for javascript errors. Looks like the momentjs is not correctly loaded

**Comment by AlbertoP57** [comment]
Chrome:
![image](https://user-images.githubusercontent.com/18015672/75764969-0071c400-5d3f-11ea-9330-bc4e55e23f03.png)

FF:
![image](https://user-images.githubusercontent.com/18015672/75764984-09629580-5d3f-11ea-94b2-85e44990c1ed.png)

moment.js seems to be included:
![image](https://user-images.githubusercontent.com/18015672/75765050-29925480-5d3f-11ea-95e5-5378b0fc27d6.png)

I havent noticed the use of moment.js instead of jquery.timeago. Now I know where to investigate. TY

**Comment by AlbertoP57** [comment]
It seems (to me) that moment.js is not jquery compliant. So it seems impossible (at least to me ...) to update "on the fly" the text arrived to the browser.

From the (YAF) file english.xml this is the description (resource) of functionality "SHOW_RELATIVE_TIME_HELP":
_If checked, client-side "Time Ago" library will be used to show "relative times" to users instead of UTC and or imperfect server-side times._

The YAF function renders a HTML code like this:

`<abbr class="timeago" title="02 March 2020 11:31:23" data-toggle="tooltip" data-html="true">2020-03-02T10:31:23Z</abbr>
`

To maintain the same behaviour you could add a jquery function like this one:

```
        jQuery(document).ready(function () {
            jQuery("abbr.timeago").each(function () {
                $(this).text(moment($(this).text()).toNow());
            });
        });

```

Furthermore moment.js should be initialized with proper locale. Example (in my case):
`moment.locale('it');`

**Comment by w8tcha** [comment]
Sorry but I don't understand the problem momentjs is loaded with the locale of the current language and correctly initialized.

**Comment by w8tcha** [comment]
I tried to reproduce the issue with the master branch. But it works as expected.

I need more info to reproduce it. Which version are you runing? Did you make any modifications? Are there still any js errors? Check the html output is the moment js actually loaded?

**Comment by AlbertoP57** [comment]
Back.

> Which version are you runing?

YAF.NET 2.32 ALPHA 202002261318 

> Did you make any modifications?

No. I never modify the core. I am just focusing on UI.

> Are there still any js errors?

No

> Check the html output is the moment js actually loaded?

I guesso so. 

Well ... I noticed that in some places it is working correctly. I studied something on jquery.timeago and moment.js. I created a simple html file to test both functions. From what I got:
The first (jquery.timeago), as the title says, is born to be used with jquery. Moment.js must be called from js code.

What is working? This:
`<span class='popover-timeago'>2020-03-02T10:31:57Z</span>`

What is NOT working? This:
`<abbr class="timeago" title="04 March 2020 13:31:23" data-toggle="tooltip" data-html="true">2020-03-04T13:22:23Z</abbr>`

The code not working comes from DisplayDateTime.cs:
https://github.com/YAFNET/YAFNET/blob/master/yafsrc/YAF.Web/Controls/DisplayDateTime.cs
![img](https://user-images.githubusercontent.com/18015672/75959777-27ef9a80-5ebf-11ea-93f8-a6d6ec12efc2.png)

called from each control:
`<YAF:DisplayDateTime`

As I said, Moment.js cannot dream of converting a text (date) without an explicit command. In fact I suggested:

```
            jQuery("abbr.timeago").each(function () {
                $(this).text(moment($(this).text()).fromNow());
            });

```
But why does it sometimes work? Investigating about "popover-timeago" the file JavaScriptBlocks.cs does this:
https://github.com/YAFNET/YAFNET/blob/2d270f8fd08b897bff88e60635e329e3d68cc100/yafsrc/YAF.Core/Utilities/JavaScriptBlocks.cs
![img1](https://user-images.githubusercontent.com/18015672/75961058-8a499a80-5ec1-11ea-9d91-507a5b64d7e4.png)

```
{Config.JQueryAlias}('.popover-timeago').each(function() {{
    {Config.JQueryAlias}(this).html(function(index, value) {{
        return moment(value).fromNow();
}}); 
```
This code is similar to what I had proposed. It does the same things.

But what for other "timeago" classes?  They are never called therefore the date remains in utc format.

This is "core" and I dont know wich strategy are you plannig to act. I cannot modify the code my way. Its up to you.

Thats all!

**Comment by w8tcha** [comment]
Okay i found the problem, fix soon!

**Comment by w8tcha** [comment]
Its causes because it was disabled on the posts page when ckeditor is used. it should work now everywhere. But there is still the transition while the editor is been changed.

**Comment by john2014** [comment]
Since its a simple fix, can this be moved to 2.31.2 release (instead of 3.0.0 release which may take a long time)?

**Comment by w8tcha** [comment]
> Since its a simple fix, can this be moved to 2.31.2 release (instead of 3.0.0 release which may take a long time)?

No its not necessary. The issue was only appearing in current alpha
